worlds/witness/player_items.py

```diff
@@ -148,39 +149,119 @@ def get_filler_items(self, quantity: int) -> Dict[str, int]:
+            good_symbols = [
+                static_witness_logic.get_parent_progressive_item(item) for item in good_symbols
+            ]
+
+            output["Symbol"] = [symbol for symbol in good_symbols if symbol in existing_items_lookup]
+
+        if "Door / Door Panel" in self._world.options.early_good_items.value:
+            good_doors = [
+                "Desert Doors & Elevator", "Keep Hedge Maze Doors", "Keep Pressure Plates Doors",
+                "Shadows Lower Doors", "Tunnels Doors",
+
+                "Keep Tower Shortcut (Door)", "Shadows Timed Door", "Tunnels Town Shortcut (Door)",
+
+                "Desert Panels", "Keep Hedge Maze Panels",
+
+                "Shadows Door Timer (Panel)", "Keep Hedge Maze 1 (Panel)", "Town Maze Stairs (Panel)",
+            ]
+
+            # While Caves Shortcuts don't unlock anything in symbol shuffle, you'd still rather have them early.
+            # But, we need to do some special handling for them.
+
+            if self._world.options.shuffle_doors in ("doors", "mixed"):
+                # Caves Shortcuts might exist in vanilla/panel doors because of "early caves: add to pool".
+                # But if the player wanted them early, they would have chosen "early caves: starting inventory".
+                # This is why we make sure these are "natural" Caves Shortcuts.
+                good_doors.append("Caves Shortcuts")
+
+            # These two doors are logically equivalent, so we choose a random one as to not give them twice the chance.
+            good_doors.append(
+                self._world.random.choice(["Caves Mountain Shortcut (Door)", "Caves Swamp Shortcut (Door)"])
+            )
+
+            if self._world.options.shuffle_vault_boxes:
+                good_doors.append("Windmill & Theater Doors")
+                if not self._world.options.shuffle_symbols:
+                    good_doors += [
+                        "Windmill & Theater Panels",
+
+                        "Windmill & Theater Control Panels",
+                    ]
+
+            if self._world.options.shuffle_doors == "doors":  # It's not as good in mixed doors because of Light Control
+                good_doors.append("Desert Light Room Entry (Door)")
+
+            if not self._world.options.shuffle_symbols:
+                good_doors += [
+                    "Bunker Doors", "Swamp Doors", "Glass Factory Doors", "Town Doors",
+
+                    "Bunker Entry (Door)", "Glass Factory Entry (Door)", "Symmetry Island Lower (Door)",
+
+                    "Bunker Panels", "Swamp Panels", "Quarry Outside Panels", "Glass Factory Panels"
+
+                    "Glass Factory Entry (Panel)",
+                ]
+
+                if self._world.options.shuffle_vault_boxes:
+                    good_doors += [
+                        "Windmill & Theater Panels",
+
+                        "Windmill & Theater Control Panels",
+                    ]
```

Comment:
Is this the same case as earlier when these can also be added?

Reply:
No

The individual "Theater Video Input (Panel)", yes
But the control panel items need to check, because you can have "Windmill & Theater Control Panels" with the videos disabled, and it just activates Windmill Turn Control

Reply:
Oh I understand now